### PR TITLE
Comment on how type hierarchy differs from maths definitions

### PR DESCRIPTION
**Pull request by IsakFalk**
Add a sentence on how the types `SemiMetric` and `Metric` does not include positive definiteness as in the mathematical definition. It is possible for an object of these types to have  `x != y` and `d(x, y) = 0`.

This follows the discussion of #153.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,7 +133,12 @@ At the top of this hierarchy is an abstract class **PreMetric**, which is define
 
 	d(x, z) <= d(x, y) + d(y, z)  for all x, y, z
 
-This type system has practical significance. For example, when computing pairwise distances between a set of vectors, you may only perform computation for half of the pairs, and derive the values immediately for the remaining halve by leveraging the symmetry of *semi-metrics*.
+This type system has practical significance. For example, when computing pairwise distances between a set of vectors, you may only
+perform computation for half of the pairs, and derive the values immediately for the remaining halve by leveraging the symmetry of
+*semi-metrics*. This is why the type definitions for `SemiMetric` and
+`Metric` do not completely follow the mathematical definition of these
+structures which include positive definiteness: `x != y` does not imply
+that `d(x, y) != 0` in general, as this does not change computations.
```
Comment (johnnychen94) [inline README.md]: Although I can read/parse it, this sentence is a little bit long and hard to read for me... I think it isn't so obvious which `which` in "which include positive definiteness" and `this` in "as this does not change computations" refer to.

Comment (IsakFalk) [reply README.md]: Thanks for the swift review, will try to rewrite more clearly.

File: README.md [hunk 2]
```diff
@@ -133,7 +133,14 @@ At the top of this hierarchy is an abstract class **PreMetric**, which is define
 
 	d(x, z) <= d(x, y) + d(y, z)  for all x, y, z
 
-This type system has practical significance. For example, when computing pairwise distances between a set of vectors, you may only perform computation for half of the pairs, and derive the values immediately for the remaining halve by leveraging the symmetry of *semi-metrics*.
+This type system has practical significance. For example, when computing pairwise distances between a set of vectors, you may only
+perform computation for half of the pairs, and derive the values immediately for the remaining halve by leveraging the symmetry of
```
Comment (dkarrasch) [inline README.md:137]:
```suggestion
perform computation for half of the pairs, and derive the values immediately for the remaining half by leveraging the symmetry of
```

**Comment by IsakFalk** [comment]
Fixed the typo and formatted the paragraph to use 92 character long lines. Not sure why the build is failing.

**Comment by dkarrasch** [comment]
I think the failure is unrelated. Let's get this in.

**Pull request merged**